jujusvg.go

```diff
@@ -13,8 +13,38 @@ import (
 // NewFromBundle returns a new Canvas that can be used
 // to generate a graphical representation of the given bundle
 // data. The iconURL function is used to generate a URL
-// that refers to an SVG for the supplied charm URL.
-func NewFromBundle(b *charm.BundleData, iconURL func(*charm.Reference) string) (*Canvas, error) {
+// that refers to an SVG for the supplied charm URL.  If
+// requested, icon SVGs that are available will be embedded
+// within the bundle SVG so that the bundle will be self-
+// contained as much as possible.  If that is to be the case,
+// utilize the provided IconFetcher (or default to HttpFetcher
+// if nil) to retrieve the icons.
+func NewFromBundle(b *charm.BundleData, iconURL func(*charm.Reference) string, embedIcons bool, fetcher IconFetcher) (*Canvas, error) {
```

Comment:
I disagree mostly because IconFetcher should allow non-url based fetching - that is, we should allow creating a fetcher for the charmstore that simply pulls the icons from the archive and populates an iconMap, wherein it doesn't make sense to have FetchIcons retrieve a URL.  I've simplified this a little in current work by removing embedIcons and having the default fetcher simply produce the `<image>` tags necessary, and having the HttpFetcher be an optional, provided interface. These changes will be up soon, so that we can talk about them.